sdk/typescript/src/types/transactions.ts

```diff
@@ -4,15 +4,16 @@
 import { ObjectOwner, SuiAddress, TransactionDigest } from './common';
 import { ObjectId, RawObjectRef } from './objects';
 
-export type Transfer = {
+export type TransferCoin = {
   recipient: string;
   object_ref: RawObjectRef;
 };
 export type RawAuthoritySignInfo = [AuthorityName, AuthoritySignature];
 
-export type TransactionKindName = 'Transfer' | 'Publish' | 'Call';
+export type TransactionKindName = 'Transfer' | 'TransferCoin' | 'Publish' | 'Call';
```

Comment:
```suggestion
export type TransactionKindName = 'TransferCoin' | 'Publish' | 'Call';
```

I would rather keep this in sync with the [backend schema](https://github.com/MystenLabs/sui/blob/7b3d8efa28a78edf34f645b940d4bee31b894733/sui_types/src/messages.rs#L84), rather than worrying about backward compatibility since this makes the codebase very messy. We will have a [devnet release](https://mysten-labs.slack.com/archives/C02GD7J9HUM/p1652375233225659) today, so the devnet gateway server should also return `TransferCoin` soon

Reply:
I would too, but i fear that may break the existing deployment - when we redeploy, will that remove all old objects ?